Readme.txt
This addon will try to switch to a tabard that provides rep when entering a dungeon.

Status: 
Seems to work pretty OK
Still lots of debug logging

Known limitations:
When the addon could not switch out the tabard when leaving the dungeon (for example 
because bags were full) then you have to fix it manually


I made the addon because I forget to switch to a rep tabard when entering a dungeon.
Normally I wear a pretty tabard or none. It picks the tabard for the faction that
has most rep already so you will finish that factions rep first and then finish another one.

So basically all the addon does is try to switch to a tabard that provides rep when in a dungeon.

Note:
This is also a playground to getting to know Git a bit.


changes.txt
Change history

1.4
- Fix applying cataclysm tabards in non-cataclysm dungeons (they don't work)
- TOC to 50400

1.3
- Fixes for 50001
- Added Mists dungeons

1.2
- TOC bump for 4.3

1.1
- TOC update for 4.2

1.0
- Initial version

Autobard.lua
--[[
This addon will try to switch to a tabard that provides rep when entering a dungeon.

Rationale:
I made the addon because I forget to switch to a rep tabard when entering a dungeon.
Normally I wear a pretty tabard or none. It picks the tabard for the faction that
has most rep already so you will finish that factions rep first and then finish another one.

Basically all the addon does is try to switch to a tabard that provides rep when in a dungeon.

]]--


local _, ATBD = ...

-- "Constants"
local MAX_REP = 42999 -- Is max rep when Exalted and the bar is full


-- Frame to catch events
ATBD.frame = CreateFrame("Frame", "AutobardFrame")

-- Defaults
--ATBD.fromTabard = nil   -- Tabard that we switched from
--ATBD.toTabard = nil     -- Tabard that we switched to



-- Calculates a number that is a measure of rep.
function ATBD.GetFactionRep(factionId)
	local name, description, standingID, barMin, barMax, barValue = GetFactionInfoByID(factionId)

	return barValue
end


-- Find out if this dungeon provides rep for the given faction
function ATBD.DungeonProvidesRepForFaction(mapId, factionId)

print("factionId", factionId)
--	local mapId = GetCurrentMapAreaID()

	print("mapId", mapId)
	-- Get factionrep group for factionID (to be defined table)
	local repGroup = ATBD.factions[factionId]
    if (not ATBD.dungeons[mapId]) then
        print("--> Unknown dungeon/mapid <--");
        return false
    end
    
print("repGroup, dungeonRepGroups:", repGroup, ATBD.dungeons[mapId].normal)

    -- Cataclysm faction tabards only provide rep while in Cataclysm dungeons
    if ((repGroup == ATBD_REPGROUP_CATA) and not(ATBD_EXP_CATA == ATBD.dungeons[mapId].xpack)) then
print("NOK Tabard only works in Cataclysm dungeons")
        return false
    end

	-- Check if correct rep in normal mode
	if (ATBD.dungeons[mapId].normal and bit.band(ATBD.dungeons[mapId].normal, repGroup) > 0) then 
print("Normal OK")
		return true 
	end

	-- Check if correct rep in heroic mode if applicable
	if (ATBD.dungeons[mapId].heroic and GetDungeonDifficulty() == 2) then  --2 means Heroic
		if (bit.band(ATBD.dungeons[mapId].heroic, repGroup) > 0) then 
print("Heroic OK")
			return true 
		end
	end

print("Not OK")
	return false
end


-- Equips best rep tabard
-- Delay it because it seems that the GetCurrentMapAreaID does not return the correct zone immediately
-- It first gives back the old zone for a while it seems.
function ATBD.EquipRepTabard()
	ATBD.delayEquipTimer = 5
	ATBD.frame:SetScript("OnUpdate", ATBD.OnUpdateDelay)
end

function ATBD.DelayedEquipRepTabard()
	local availableItems = {}
	GetInventoryItemsForSlot(INVSLOT_TABARD, availableItems)

	if (not availableItems) then
		-- Abort if no tabards available
		return
	end

	-- See if there is a tabard that gives rep in this dungeon
	SetMapToCurrentZone()
	local mapId = GetCurrentMapAreaID()
	local currentTabardId = GetInventoryItemID("player", INVSLOT_TABARD)
	local lastRep = 0
	local bestTabard

print("Current: ", currentTabardId)

	for _, tabardId in pairs(availableItems) do
print("tabardId: ", tabardId)
		if (ATBD.tabards[tabardId]) then
print("tabardId known: ", tabardId)

			if ( ATBD.DungeonProvidesRepForFaction(mapId, ATBD.tabards[tabardId])) then

				-- Its a different tabard and one that the addon knows (so it will provide rep)
				local thisRep = ATBD.GetFactionRep(ATBD.tabards[tabardId])
	print("rep: ", thisRep)

				if ((thisRep < MAX_REP) and (thisRep > lastRep)) then
					-- If this faction has more rep, prefer this one
					bestTabard = tabardId
					lastRep = thisRep
	print("bestTabard: ", bestTabard)
				end
			end
		end
	end

	-- Equip the tabard (if one is found and it is a different one)
	if (bestTabard and bestTabard ~= currentTabardId) then
		ATBD.fromTabard = currentTabardId
		ATBD.toTabard = bestTabard

print("From, To", ATBD.fromTabard, ATBD.toTabard)

		EquipItemByName(ATBD.toTabard) -- Also works with IDs instead of names

		local equippedTabard = GetInventoryItemID("player", INVSLOT_TABARD)
		if (equippedTabard ~= ATBD.toTabard) then
			print("Equipping tabard failed, retry...")
			ATBD.retryTimer = 3
			ATBD.frame:SetScript("OnUpdate", ATBD.OnUpdateRetry)
		end
	end

end


-- Dequips rep tabard if it was put on by the addon
function ATBD.DequipRepTabard()
	if (not UnitIsDeadOrGhost("player")) then
		-- We are not dead
print("Not dead")

		if (ATBD.toTabard) then
			-- We changed the players tabard

			print("Switch back")
			if (IsEquippedItem(ATBD.toTabard)) then
				-- Player is still wearing the switched to tabard
				-- So try to switch back
				if (ATBD.fromTabard) then
print("Back from, To", ATBD.toTabard, ATBD.fromTabard)
					EquipItemByName(ATBD.fromTabard) -- Also works with IDs
				else
print("Dequipping", ATBD.toTabard)
					-- No tabard to switch back to, so dequip the current one
					PickupInventoryItem(INVSLOT_TABARD)

					-- This is silly, I can either put it in the Backpack or another Bag
					-- there seems to be no function just put it in any bag
					-- So manually figure out if there is space in bag or backpack
					for containerId = NUM_BAG_SLOTS, 1, -1 do
						freeSlots, bagType = GetContainerNumFreeSlots(containerId)
print("Baginfo id, free, type", containerId, freeSlots, bagType)
						if  ((freeSlots > 0) and (bagType == 0)) then
print("Putting it in bag", containerId, ContainerIDToInventoryID(containerId))
							PutItemInBag(ContainerIDToInventoryID(containerId))  -- This mapping seems to make it work
							break;
						end
					end

					-- If it is still on the cursor attempt the backpack
					-- when that is also full the player will get the UI error
					-- that his bags are all full, which is OK
					if (CursorHasItem()) then
print("Attempting backpack")
						PutItemInBackpack()
					end
				end

				ATBD.toTabard = nil
			end
		end
	end
end

-- Entering world see if we are in a dungeon or not and take action
function ATBD.PLAYER_ENTERING_WORLD(self, event, ...)
print("PLAYER_ENTERING_WORLD")

	local inInstance, instanceType = IsInInstance()

	if (inInstance and instanceType == "party") then
		-- We are in a 5 man instance
		if (InCombatLockdown()) then
			-- Try later when the fight is over
			ATBD.delayedEnteringWorld = true
			ATBD.frame:RegisterEvent("PLAYER_REGEN_ENABLED")
		else
			ATBD.EquipRepTabard()
		end
	else
		-- Not in a dungeon
		ATBD.DequipRepTabard()
	end
end


-- Player leaving world, restore previous tabard
function ATBD.PLAYER_LEAVING_WORLD(self, event, ...)
print("PLAYER_LEAVING_WORLD")
		if (not InCombatLockdown()) then
			-- Don't bother to try later, 
			--   we will either enter te world again
			--   or the player is logging out
			ATBD.DequipRepTabard()
		end
end


-- Player reputation changed, see if we can switch tabard
function ATBD.UPDATE_FACTION(self, event, ...)
--print("UPDATE_FACTION")

	if (InCombatLockdown()) then
		-- Check later
		ATBD.delayedUpdateFaction = true
		ATBD.frame:UnregisterEvent("UPDATE_FACTION")
		ATBD.frame:RegisterEvent("PLAYER_REGEN_ENABLED")
		return
	end

	if (ATBD.toTabard) then
		-- We changed the players tabard
		if (IsEquippedItem(ATBD.toTabard)) then
			-- Player is still wearing it

			if (ATBD.GetFactionRep(ATBD.tabards[ATBD.toTabard]) >= MAX_REP) then
				-- We got max rep, switch tabard
				print("**** MAX Rep, switching tabard ****")
-- Dequipping is automatic				ATBD.DequipRepTabard()
				ATBD.EquipRepTabard()
			end
		end
	end
end


-- Player got out of combat, see if we can switch tabard now if it was delayed
function ATBD.PLAYER_REGEN_ENABLED(self, event, ...)
print("PLAYER_REGEN_ENABLED")
	if (ATBD.delayedUpdateFaction) then
		ATBD.UPDATE_FACTION(self, event, ...)
		ATBD.frame:RegisterEvent("UPDATE_FACTION")
	end

	if (ATBD.delayedEnteringWorld) then
		ATBD.PLAYER_ENTERING_WORLD(self, event, ...)
	end

	ATBD.frame:UnregisterEvent("PLAYER_REGEN_ENABLED")
end


-- Retry equipping after a while
function ATBD.OnUpdateDelay(self, elapsed)
	ATBD.delayEquipTimer = ATBD.delayEquipTimer - elapsed

	if (ATBD.delayEquipTimer < 0) then
		ATBD.frame:SetScript("OnUpdate", nil)
		ATBD.DelayedEquipRepTabard()
	end

end

function ATBD.OnUpdateRetry(self, elapsed)
	ATBD.retryTimer = ATBD.retryTimer - elapsed

	if (ATBD.retryTimer < 0) then
		if(ATBD.toTabard) then
			local equippedTabard = GetInventoryItemID("player", INVSLOT_TABARD)

			if (equippedTabard ~= ATBD.toTabard) then
				print("Tabard still not equipped, equip it")
				-- Retry equipping tabard
				EquipItemByName(ATBD.toTabard) -- Also works with IDs instead of names

				equippedTabard = GetInventoryItemID("player", INVSLOT_TABARD)
				if (equippedTabard ~= ATBD.toTabard) then
					print("Equipping tabard failed again, retry...")
					ATBD.retryTimer = 5
					ATBD.frame:SetScript("OnUpdate", ATBD.frame.OnUpdateRetry)
				else
					print("Equipping tabard succeeded")
					ATBD.frame:SetScript("OnUpdate", nil)
				end
			else
				print("Tabard got equipped in the meantime")
				ATBD.frame:SetScript("OnUpdate", nil)
			end
		else
			print("No tabard to equip anymore")
			ATBD.frame:SetScript("OnUpdate", nil)
		end
	end
end

-- Generic event dispatcher
ATBD.frame:SetScript("OnEvent", function(self, event, ...) ATBD[event](self, event, ...) end )

-- Events
ATBD.frame:RegisterEvent("PLAYER_ENTERING_WORLD")
ATBD.frame:RegisterEvent("PLAYER_LEAVING_WORLD")
ATBD.frame:RegisterEvent("UPDATE_FACTION")



TabardList.lua
--[[
List of tabards and for which faction they provide rep
]]--

local _, ATBD = ...

-- Mapping table of tabard ID's to the faction that they give rep for
local tl = {}

-- Alliance cities
tl[45574] = 72			-- Stormwind
tl[45577] = 47			-- Ironforge
tl[45578] = 54			-- Gnomeregan
tl[45579] = 69			-- Darnassus
tl[45580] = 930			-- Exodar
tl[64882] = 1134		-- Gilneas

-- Horde cities
tl[45581] = 76			-- Orgrimmar
tl[45582] = 530			-- Darkspear Trolls
tl[45583] = 68			-- Undercity
tl[45584] = 81			-- Thunder Bluff
tl[45585] = 911			-- Silvermoon City
tl[64884] = 1133		-- Bilgewater Cartel

-- Lich King factions
tl[43154] = 1106		-- Argent Crusade
tl[43155] = 1098		-- Knights of the Ebon Blade
tl[43156] = 1091		-- The Wyrmrest Accord
tl[43157] = 1090		-- Kirin'Tor

-- Cata factions
tl[65905] = 1135		-- The Earthen Ring
tl[65906] = 1158		-- Guardians of Hyjal
tl[65907] = 1171		-- Therazane
tl[65909] = 1172		-- Dragonmaw Clan
tl[65904] = 1173		-- Ramkahen
tl[65908] = 1174		-- Wildhammer Clan

-- Mist factions
tl[83079] = 1353		-- Tushui Pandaren

-- Guild, there are more tabard that give rep for this faction
-- I assume player will only carry one so no special stuff needed
tl[69209] = 1168		-- Illustrious Guild Tabard 
tl[69210] = 1168		-- Renowned Guild Tabard (epic one)



ATBD.tabards = tl


FactionList.lua
--[[
List of factions to repgroups
]]--

local _, ATBD = ...

-- Mapping table of tabard ID's to the faction that they give rep for
local fl = {}

-- Alliance cities
fl[72]	 = ATBD_REPGROUP_CITY	-- Stormwind
fl[47]	 = ATBD_REPGROUP_CITY	-- Ironforge
fl[54]	 = ATBD_REPGROUP_CITY	-- Gnomeregan
fl[69]	 = ATBD_REPGROUP_CITY	-- Darnassus
fl[930]	 = ATBD_REPGROUP_CITY	-- Exodar
fl[1134] = ATBD_REPGROUP_CITY	-- Gilneas

-- Horde cities
fl[76]	 = ATBD_REPGROUP_CITY-- Orgrimmar
fl[530]	 = ATBD_REPGROUP_CITY-- Darkspear Trolls
fl[68]	 = ATBD_REPGROUP_CITY-- Undercity
fl[81]	 = ATBD_REPGROUP_CITY-- Thunder Bluff
fl[911]	 = ATBD_REPGROUP_CITY-- Silvermoon City
fl[1133] = ATBD_REPGROUP_CITY-- Bilgewater Cartel

-- Lich King factions
fl[1106] = ATBD_REPGROUP_WOTLK -- Argent Crusade
fl[1098] = ATBD_REPGROUP_WOTLK -- Knights of the Ebon Blade
fl[1091] = ATBD_REPGROUP_WOTLK -- The Wyrmrest Accord
fl[1090] = ATBD_REPGROUP_WOTLK -- Kirin'Tor

-- Cata factions
fl[1135] = ATBD_REPGROUP_CATA -- The Earthen Ring
fl[1158] = ATBD_REPGROUP_CATA -- Guardians of Hyjal
fl[1171] = ATBD_REPGROUP_CATA -- Therazane
fl[1172] = ATBD_REPGROUP_CATA -- Dragonmaw Clan
fl[1173] = ATBD_REPGROUP_CATA -- Ramkahen
fl[1174] = ATBD_REPGROUP_CATA -- Wildhammer Clan

-- Mists factions
fl[1352] = ATBD_REPGROUP_MISTS -- Huojin Pandaren
fl[1353] = ATBD_REPGROUP_MISTS -- Tushui Pandaren

-- Guild faction
fl[1168] = ATBD_REPGROUP_ALL -- Illustrious Guild Tabard 
fl[1168] = ATBD_REPGROUP_ALL -- Renowned Guild Tabard (epic one)



ATBD.factions = fl


DungeonList.lua
--[[
List of dungeons with their associated "repgroups"
]]--

local _, ATBD = ...

-- Dungeonlist
-- Note that Heroic only needs to list additional reps compared to normal

local dl = {}


-- *** Classic dungeons ***

dl[680] = { normal = ATBD_REPGROUP_CITY } -- RagefireChasm
dl[686] = { normal = ATBD_REPGROUP_CITY } -- Zul'farrak
dl[687] = { normal = ATBD_REPGROUP_CITY } -- Sunken Temple
dl[688] = { normal = ATBD_REPGROUP_CITY } -- Black Fathom Deeps
dl[690] = { normal = ATBD_REPGROUP_CITY } -- Stockades
dl[691] = { normal = ATBD_REPGROUP_CITY } -- Gnomeregan
dl[692] = { normal = ATBD_REPGROUP_CITY } -- Uldaman
dl[699] = { normal = ATBD_REPGROUP_CITY } -- DireMaul
dl[704] = { normal = ATBD_REPGROUP_CITY } -- Blackrock Depths
dl[721] = { normal = ATBD_REPGROUP_CITY } -- BlackrockSpire
dl[749] = { normal = ATBD_REPGROUP_CITY } -- WailingCaverns
dl[750] = { normal = ATBD_REPGROUP_CITY } -- Maraudon
dl[760] = { normal = ATBD_REPGROUP_CITY } -- Razorfen Downs
dl[761] = { normal = ATBD_REPGROUP_CITY } -- Razorfen Kraul
dl[762] = { normal = ATBD_REPGROUP_CITY } -- Scarlet Monastry
dl[763] = { normal = ATBD_REPGROUP_CITY } -- Scholomance
dl[765] = { normal = ATBD_REPGROUP_CITY } -- Stratholme


-- *** BC dungeons ***
-- These provide no rep through tabards (yet?)

dl[710] = { } -- The Shattered Halls
dl[797] = { } -- Hellfire Ramparts
dl[725] = { } -- The Blood Furnace

dl[726] = { } -- The Underbog
dl[727] = { } -- The Steamvault
dl[728] = { } -- The Slave Pens

dl[724] = { } -- Shadow Labyrinth
dl[732] = { } -- Mana-Tombs
dl[722] = { } -- Auchenai Crypts
dl[723] = { } -- Sethekk Halls

dl[729] = { } -- The Botanica
dl[730] = { } -- The Mechanar
dl[731] = { } -- The Arcatraz

dl[734] = { } -- CoT: Old Hillsbrad
dl[733] = { } -- CoT: The Black Morass

dl[798] = { } -- Magister's Terrace


-- *** WotLK Dungeons ***

dl[523] = { normal = ATBD_REPGROUP_CITY, heroic = ATBD_REPGROUP_WOTLK }  -- Utgarde Keep
dl[520] = { normal = ATBD_REPGROUP_CITY, heroic = ATBD_REPGROUP_WOTLK }  -- The Nexus
dl[533] = { normal = ATBD_REPGROUP_CITY, heroic = ATBD_REPGROUP_WOTLK }  -- Azjol-Nerub: Azjol-Nerub
dl[522] = { normal = ATBD_REPGROUP_CITY, heroic = ATBD_REPGROUP_WOTLK }  -- Ahn'kahet: The Old Kingdom
dl[534] = { normal = ATBD_REPGROUP_CITY, heroic = ATBD_REPGROUP_WOTLK }  -- Drak'Tharon Keep
dl[536] = { normal = ATBD_REPGROUP_CITY, heroic = ATBD_REPGROUP_WOTLK }  -- The Violet Hold
dl[530] = { normal = ATBD_REPGROUP_CITY, heroic = ATBD_REPGROUP_WOTLK }  -- Gundrak
dl[526] = { normal = ATBD_REPGROUP_CITY, heroic = ATBD_REPGROUP_WOTLK }  -- Halls of Stone
dl[528] = { normal = ATBD_REPGROUP_CITY + ATBD_REPGROUP_WOTLK }  -- Oculus
dl[525] = { normal = ATBD_REPGROUP_CITY + ATBD_REPGROUP_WOTLK }  -- Halls of Lightning
dl[524] = { normal = ATBD_REPGROUP_CITY + ATBD_REPGROUP_WOTLK }  -- Utgarde Pinnacle
dl[521] = { normal = ATBD_REPGROUP_CITY + ATBD_REPGROUP_WOTLK }  -- Culling of Stratholme
dl[542] = { normal = ATBD_REPGROUP_CITY + ATBD_REPGROUP_WOTLK }  -- Trial of the Champion
dl[601] = { normal = ATBD_REPGROUP_CITY + ATBD_REPGROUP_WOTLK }  -- Forge of Souls
dl[602] = { normal = ATBD_REPGROUP_CITY + ATBD_REPGROUP_WOTLK }  -- Pit of Saron
dl[603] = { normal = ATBD_REPGROUP_CITY + ATBD_REPGROUP_WOTLK }  -- Halls of Reflection

-- *** Cataclysm dungeons ***

dl[756] = { xpack = ATBD_EXP_CATA, normal = ATBD_REPGROUP_CITY, heroic = ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA }   -- The Deadmines
dl[764] = { xpack = ATBD_EXP_CATA, normal = ATBD_REPGROUP_CITY, heroic = ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA }   -- Shadowfang Keep
dl[753] = { xpack = ATBD_EXP_CATA, normal = ATBD_REPGROUP_CITY  + ATBD_REPGROUP_WOTLK, heroic = ATBD_REPGROUP_CATA }  -- Blackrock Caverns
dl[767] = { xpack = ATBD_EXP_CATA, normal = ATBD_REPGROUP_CITY  + ATBD_REPGROUP_WOTLK, heroic = ATBD_REPGROUP_CATA }  -- Throne of Tides
dl[768] = { xpack = ATBD_EXP_CATA, normal = ATBD_REPGROUP_CITY  + ATBD_REPGROUP_WOTLK, heroic = ATBD_REPGROUP_CATA }  -- Stonecore
dl[769] = { xpack = ATBD_EXP_CATA, normal = ATBD_REPGROUP_CITY  + ATBD_REPGROUP_WOTLK, heroic = ATBD_REPGROUP_CATA }  -- Vortex Pinnacle
dl[759] = { xpack = ATBD_EXP_CATA, normal = ATBD_REPGROUP_CITY  + ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA }          -- GrimBatol
dl[759] = { xpack = ATBD_EXP_CATA, normal = ATBD_REPGROUP_CITY  + ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA }          -- Halls of Origination
dl[747] = { xpack = ATBD_EXP_CATA, normal = ATBD_REPGROUP_CITY  + ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA }          -- Lost City of Tol'vir

-- *** Patch 4.1 dungeons ***
dl[793] = { xpack = ATBD_EXP_CATA, normal = ATBD_REPGROUP_CITY  + ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA }          -- Zul'Gurub
dl[781] = { xpack = ATBD_EXP_CATA, normal = ATBD_REPGROUP_CITY  + ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA }          -- Zul'Aman

-- *** Patch 4.3 dungeons ***
dl[820] = { xpack = ATBD_EXP_CATA, normal = ATBD_REPGROUP_CITY  + ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA }          -- End Time
dl[816] = { xpack = ATBD_EXP_CATA, normal = ATBD_REPGROUP_CITY  + ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA }          -- Well of Eternity
dl[819] = { xpack = ATBD_EXP_CATA, normal = ATBD_REPGROUP_CITY  + ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA }          -- Hour of Twilight


-- *** Patch 5.0.4 dungeons ***
dl[871] = { normal = ATBD_REPGROUP_CITY } -- Scarlett Halls

-- *** Mists dungeons *** THE REPS ARE GUESSES FOR NOW
dl[875] = { normal = ATBD_REPGROUP_CITY + ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA + ATBD_REPGROUP_MISTS } -- Gate of the Setting Sun MoP
dl[885] = { normal = ATBD_REPGROUP_CITY + ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA + ATBD_REPGROUP_MISTS } -- Mogu'Shan Palace MoP
dl[877] = { normal = ATBD_REPGROUP_CITY + ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA + ATBD_REPGROUP_MISTS } -- Shado-pan Monastery MoP
dl[887] = { normal = ATBD_REPGROUP_CITY + ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA + ATBD_REPGROUP_MISTS } -- Siege of Niuzao Temple MoP
dl[876] = { normal = ATBD_REPGROUP_CITY + ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA + ATBD_REPGROUP_MISTS } -- Stormstout Brewery MoP
dl[867] = { normal = ATBD_REPGROUP_CITY + ATBD_REPGROUP_WOTLK + ATBD_REPGROUP_CATA + ATBD_REPGROUP_MISTS } -- Temple of the Jade Serpent 

ATBD.dungeons = dl
